src/components/structures/SpaceRoomView.tsx

```diff
@@ -672,25 +671,11 @@ export default class SpaceRoomView extends React.PureComponent<IProps, IState> {
 
         if (payload.action !== Action.ViewUser && payload.action !== "view_3pid_invite") return;
 
-        if (payload.action === Action.ViewUser && payload.member) {
-            const spaceMemberInfoCard: IRightPanelCard = {
-                phase: RightPanelPhases.SpaceMemberInfo,
-                state: { spaceId: this.props.space.roomId, member: payload.member },
-            };
-            if (payload.push) {
-                RightPanelStore.instance.pushCard(spaceMemberInfoCard);
-            } else {
-                RightPanelStore.instance.setCards([
-                    { phase: RightPanelPhases.SpaceMemberList, state: { spaceId: this.props.space.roomId } },
-                    spaceMemberInfoCard,
-                ]);
-            }
-        } else if (payload.action === "view_3pid_invite" && payload.event) {
-            RightPanelStore.instance.setCard({
-                phase: RightPanelPhases.Space3pidMemberInfo,
-                state: { spaceId: this.props.space.roomId, memberInfoEvent: payload.event },
-            });
-        } else {
+        /**
+         * The rest of the `ViewUser` and `view_3pid_invite` exists in the `<RoomView />`
+         * component. This deals specifically with showing the space members list
+         */
+        if (!payload.member && !payload.event) {
             RightPanelStore.instance.setCard({
                 phase: RightPanelPhases.SpaceMemberList,
                 state: { spaceId: this.props.space.roomId },
```

Comment:
I don't really know to arrange for one of these actions to happen when a SpaceRoomView is open, so I can't test it, but now that RoomView also handles these actions, won't we see flickering between the two states? Or, worse, they'll happen in the wrong order and we'll get stuck with the wrong one?

Reply:
Remove this code from `SpaceRoomView` in https://github.com/matrix-org/matrix-react-sdk/pull/11495/commits/f7fa9facd78760b21a5021df7219eab39d5e9208. [This code](https://github.com/matrix-org/matrix-react-sdk/pull/11495/files#diff-13590ebbbcd62256ff3ead94de2e6be41867b82ad9e0470c4b63ff63c1f5f78bR1270) in RoomView will do the same thing, even for a space room.